### PR TITLE
Better error messaging for invalid files

### PR DESCRIPTION
**Pull request by alexdechaves**
Screen Shot 2020-10-01 at 6 40 40 PM

### CODE REVIEW

**Review by sstriegs — APPROVED** [review]

File: pages/assets/[id].tsx [hunk 1]
```diff
@@ -49,14 +49,19 @@ const Asset: React.FC<Props> = () => {
...
+    errorMessage = {
+      initial: 'Error creating this asset: Please upload a valid video file (gifs are not supported)',
+      apiError: message
+  }}
 
   if (errorMessage) {
     return (
       <Layout darkMode={false}>
-        <div><h1>{errorMessage}</h1></div>
+        <div><h1>{errorMessage.initial}</h1></div>
```
Comment (dylanjha) [inline pages/assets/[id].tsx:60]: There's a view other places where set set `errorMessage =` to a string (line 28 and line 32). Honestly I would expect the typescript compiler to catch this. I'm wondering why it didn't.

I suggest

`errorMessage = ` is always set to a string and it's shown in the `<h1>`
`errorDetails=` is a different string that only gets set when the asset is errored. And then do something like this:

```tsx
{errorDetails && <p>{errorDetails}</p>}
```

lmk what you think of that

**Review by dylanjha — CHANGES_REQUESTED** [review]

File: pages/assets/[id].tsx (hunk 1 above)
Comment (alexdechaves) [reply pages/assets/[id].tsx:60]: yea makes sense!

File: pages/assets/[id].tsx [hunk 2]
```diff
@@ -48,14 +49,15 @@ const Asset: React.FC<Props> = () => {
...
-    errorMessage = `Error creating this asset: ${message}`;
+    errorMessage = 'Error creating this asset: Please upload a valid video file (gifs are not supported)'
+    errorDetails = asset.errors && asset.errors.messages[0];
   }
 
   if (errorMessage) {
     return (
       <Layout darkMode={false}>
         <div><h1>{errorMessage}</h1></div>
+        {errorMessage && <p>{errorDetails}</p>}
```
Comment (dylanjha) [inline pages/assets/[id].tsx:60]: @alexdechaves whoops, I just caught this, it should be:

```
{errorDetails && <p>{errorDetails}</p>}
```

Comment (alexdechaves) [reply pages/assets/[id].tsx:60]: i literally changed it back from what you gave me :(

**Pull request merged**